src/core/Akka.Remote.TestKit/Player.cs

```diff
@@ -544,53 +546,98 @@ class PlayerHandler : IHeliosConnectionHandler
             Reconnect();
         }
 
-        public void OnException(Exception ex, IConnection erroredChannel)
+        private static string FormatConnectionFailure(IChannelHandlerContext context, Exception exception)
         {
-            _log.Debug("channel {0} exception {1}", erroredChannel, ex);
-            if (ex is HeliosConnectionException && _reconnects > 0)
+            var sb = new StringBuilder();
+            sb.AppendLine($"Connection between [Local: {context.Channel.LocalAddress}] and [Remote: {context.Channel.RemoteAddress}] has failed.");
+            sb.AppendLine($"Cause: {exception}");
+            sb.AppendLine($"Trace: {exception.StackTrace}");
+            return sb.ToString();
+        }
+
+        public override void ExceptionCaught(IChannelHandlerContext context, Exception exception)
+        {
+            _log.Debug("channel {0} exception {1}", context.Channel, exception);
+            if (exception is ConnectException && _reconnects > 0)
             {
                 _reconnects -= 1;
-                _scheduler.Advanced.ScheduleOnce(_nextAttempt.TimeLeft, Reconnect);
+                if (_nextAttempt.IsOverdue)
+                {
+                    Reconnect();
+                }
+                else
+                {
+                    _scheduler.Advanced.ScheduleOnce(_nextAttempt.TimeLeft, Reconnect);
+                }
                 return;
             }
-            _fsm.Tell(new ClientFSM.ConnectionFailure(ex.ToString()));
+            _fsm.Tell(new ClientFSM.ConnectionFailure(FormatConnectionFailure(context, exception)));
         }
 
         private void Reconnect()
         {
+            _log.Debug("Connecting...");
             _nextAttempt = Deadline.Now + _backoff;
-            RemoteConnection.CreateConnection(Role.Client, _server, _poolSize, this);
+            RemoteConnection.CreateConnection(Role.Client, _server, _poolSize, this).ContinueWith(tr =>
```

Comment:
Recursive async retry routine for when initial attempts to connect to the `TestConductor` fail, usually as a result of a non-controller node starting faster than the controller itself.